package/ui/utils.py

```diff
@@ -0,0 +1,346 @@
+    heron_file = xml_file.with_suffix(".heron")
+    print(f"Converting {xml_file} to {heron_file} using {xml2eddi_script}...")
+    # Use Workbench's entry.bat script (or entry.sh) to access the app's internal Python environment
+    # to run the script
+    entry_script = workbench_path / "rte" / "entry.bat"
+    if platform.system() != "Windows":
+        entry_script = entry_script.with_suffix(".sh")
+    if not entry_script.exists():
+        print("ERROR: Could not find the entry script in the Workbench installation directory.")
+        return None
+
+    with open(heron_file, "w") as f:
+        subprocess.run([str(entry_script), str(xml2eddi_script), str(xml_file)], stdout=f)
+
+    return heron_file
+
+
+def run_in_workbench(file: str | None = None):
+    """
+    Opens the given file in the NEAMS Workbench
+    @ In, file, str, optional, the file to open in NEAMS Workbench
+    """
+    # Find the Workbench executable
+    workbench_path = find_workbench()
+    if workbench_path is None:
+        print("ERROR: Could not find the NEAMS Workbench executable. Please set the "
+              "WORKBENCH_PATH environment variable, add it to the system path, or specify it manually "
+              "with the WORKBENCHDIR key in the \".workbench\" file in the main FORCE directory.")
+        return
+
+    # Create Workbench default configuration for HERON if a default configurations file does not exist
+    workbench_install_dir = get_workbench_dir_from_exe_path(workbench_path)
+    create_workbench_heron_default(workbench_install_dir)
+
+    # Convert the .xml file to a .heron file if one was provided
+    if file is not None:
+        file = pathlib.Path(file)
+        if not file.exists():
+            print(f"ERROR: The file {file} does not exist.")
+            return
+
+        if file.suffix == ".xml":
+            heron_file = convert_xml_to_heron(file, workbench_install_dir)
+            if heron_file is None:
+                return
+            file = heron_file
+        file = str(file)
+
+    # Open the file in Workbench
+    # Currently, we're only able to open the MacOS version of Workbench by opening the app itself.
+    # This does not accept a file as an argument, so users will need to open Workbench, then open
+    # the desired file manually from within the app.
+    # if file is not None and platform.system() == "Windows":
+    #     command += ' ' + file
+
+    print("Opening Workbench...", file=sys.__stdout__)
+    print("***If this is the first time you are running Workbench, this may take a few minutes!***\n",
+          file=sys.__stdout__)
+    if platform.system() == "Windows":
+        if file is None:
+            command = str(workbench_path)
+        else:
+            command = [str(workbench_path), file]
+        print("using command", command)
+        subprocess.run(command)
+    else:
+        # NOTE: untested on Linux as of 2024-07-22
```

Comment:
Hm, I am guessing that in Linux the correct way to do this would be the same as in Windows (as in subprocess.run with the workbench executable)

Reply:
I'll edit so that the snippet being run for Windows is also run if the user is on Linux (though still untested)